pyfr/plugins/turbulencegenerator.py

```diff
@@ -53,6 +53,11 @@ class TurbulenceGeneratorPlugin(BasePlugin):
     def __init__(self, intg, cfgsect, suffix):
         super().__init__(intg, cfgsect, suffix)
 
+        # do not allow multiple instances
+        if suffix:
```

Comment:
Think generically!  This should be handled by a mechanism by which plugins can declare themselves to be singletons.  For example 

```
class MyPlugin(BasePlugin):
    name = ...
    ...
    singleton = True
```

when plugins are being constructed this property can be queried.  Also note that the current code will break if you specify only one instance but opt to include a suffix.